com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -150,7 +150,10 @@ public boolean supportsRealtimeBreakpointVerification() {
                 ValidBreakpointLocationLocator locator = new ValidBreakpointLocationLocator(astUnit, lines[i], true, true);
                 astUnit.accept(locator);
                 // When the final valid line location is same as the original line, that represents it's a valid breakpoint.
-                if (lines[i] == locator.getLineLocation()) {
+                // Add location type check to avoid breakpoint on method/field which will never be hit in current implementation.
+                if (locator.getLocationType() == ValidBreakpointLocationLocator.LOCATION_LINE) {
+                    // update lines to the next line which is valid for breakpoint
+                    lines[i] = locator.getLineLocation();
```

Comment:
Modifying what's passed in is not recommended. Please add an override to accept an extra parameter for the modified line numbers.